[Page Templates Feature] Added EuiPageSection and updated EuiPage & EuiPageBody padding defaults

cchaos opened the pull request:
**This PR's specific commits starts at:** https://github.com/elastic/eui/pull/5769/commits/cd2c51205b94ccc9cee905a06e02cf380b91a52b
**The EuiPageTemplate are slightly broken at this point, don't look there yet 🙈** 

## [Breaking] Changed default `padding` of `EuiPage` from `m` to `none`

- And removed nested page header styles when within EuiPageBody (improved way of doing this coming later)

## [[NEW] EuiPageSection to replace EuiPageContent & EuiPageContentBody](https://github.com/elastic/eui/commit/cd2c51205b94ccc9cee905a06e02cf380b91a52b)

This component handles all the `restrictWidth` and border options for page sections to ensure proper extension of the border. 

<img width="1012" alt="Screen Shot 2022-04-10 at 23 50 20 PM" src="https://user-images.githubusercontent.com/549577/162662298-870bc59d-fd2c-46ca-b35d-67863aadddc5.png">

## Deprecated EuiPageContent & EuiPageContentBody

<img width="629" alt="Screen Shot 2022-04-11 at 00 31 24 AM" src="https://user-images.githubusercontent.com/549577/162665559-ce2aee1c-b9b9-46db-b633-63ba3f6cf747.png">


## [NEW] Added `euiPaddingStyles` and `euiBackgroundColorStyles` global styling functions

I've added these new helpers in an attempt to consolidate any component's usage of padding or background colors to use a single global style option. Right now these require passing in the EuiTheme, but once #5754 merges, we'll have an easy way to create a hook version too. (That's also when we'll publicly document it).

**Basic usage:**

```tsx
const inlinePadding = euiPaddingStyles(useTheme, 'inline');
const blockPadding = euiPaddingStyles(useTheme, 'block');
const colors = euiBackgroundColorStyles(useTheme);
```


## Docs 

- Updated Page section to include page body and sidebar descriptions
- Added a section for new `Page sections`
- Added a section about complex configurations of page components

### Checklist

- [x] Checked in both **light and dark** modes
- [x] Checked in **mobile**
- [x] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
- [x] Props have proper **autodocs** and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#adding-playground-toggles)**
- [x] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md)**
- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples
- [x] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/cypress-testing.md) tests**
- [x] Checked for **breaking changes** and labeled appropriately
- [ ] Checked for **accessibility** including keyboard-only and screenreader modes
- [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart
- [x] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5769/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5769/

cchaos commented:
This feature branch PR is ready for review!

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5769/

miukimiu commented on src/components/page/page_section/page_section.tsx:
```diff
@@ -0,0 +1,104 @@
+        bottomBorder === 'extended' && styles.border,
+        // @ts-ignore HELP
+        alignment && styles[alignment],
+        // @ts-ignore HELP
+        colors[color],
```
The error goes away if you merge `main`.

miukimiu reviewed (approved):
LGTM! :tada:

Tested in Chrome, Safari, Edge, and Firefox. Found some small issues but nothing that prevents it from merging.

miukimiu commented:
@cchaos I juts noticed that because we're now using the **EuiPageSection** in `src-docs/src/components/guide_page/guide_page.js` and `src-docs/src/components/guide_tabbed_page/guide_tabbed_page.tsx` the CI is failing with: 
[page must have one main landmark](https://dequeuniversity.com/rules/axe/4.1/landmark-one-main?application=axe-puppeteer).

cchaos commented on src-docs/src/views/page/page_example.js:
```diff
@@ -47,9 +61,192 @@ export const PageExample = {
+            <p>
+              <strong>EuiPageSection</strong> is a stackable component that is
+              essentially an <strong>EuiPanel</strong> with props for quickly
+              creating common usages. Use <EuiCode>panelled</EuiCode> to quickly
```
Thank you, I originally had that in there, then I removed it.

cchaos commented:
Thanks @miukimiu ! I've addressed your comments. As for the `role` issue, I'm trying to decide whether we require each EuiPageSection to contain a `role` attribute or make `EuiPageHeader` actually render a div by default with the ability to change the `role`. For now I've added `role="main"` to the docs' EuiPageBody and removed the docs' EuiPageHeader's role with `role="none"` since you can't have nested roles.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_5769/
